### PR TITLE
[numpy] add op flipud, fliplr

### PR DESCRIPTION
**Pull request by Yiyan66**
## Description ##
add op flipud, fliplr

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/numpy/multiarray.py [hunk 1]
```diff
@@ -1469,6 +1469,22 @@ def flip(self, *args, **kwargs):
         """
         raise AttributeError('mxnet.numpy.ndarray object has no attribute flip')
 
+    def flipud(self, *args, **kwargs):
```
Comment (haojin2) [inline python/mxnet/numpy/multiarray.py]: You don't have to explicitly add those convenient methods

File: python/mxnet/ndarray/numpy/_op.py [hunk 2]
```diff
@@ -4741,6 +4742,120 @@ def flip(m, axis=None, out=None):
...
+    -----
+    Equivalent to ``m[::-1,...]``.
+    Does not require the array to be two-dimensional.
+
+    Examples
+    --------
+    >>> A = np.diag(np.array([1.0, 2, 3]))
+    >>> A
+    array([[1.,  0.,  0.],
+        [0.,  2.,  0.],
```
Comment (reminisce) [inline python/mxnet/ndarray/numpy/_op.py]: Alignment. Same for the other places.
```
array([[1.,  0.,  0.],
       [0.,  2.,  0.],
```

File: python/mxnet/ndarray/numpy/_op.py [hunk 3]
```diff
@@ -4741,6 +4742,120 @@ def flip(m, axis=None, out=None):
...
+    >>> np.flipud(np.array([1,2]))
+    array([2., 1.])
+    """
+    from ...numpy import ndarray
+    if isinstance(m, numeric_types):
+        return _np.flip(m, 0)
+    elif isinstance(m, ndarray):
+        return _npi.flip(m, 0)
+    else:
+        raise TypeError('type {} not supported'.format(str(type(m))))
```
Comment (reminisce) [inline python/mxnet/ndarray/numpy/_op.py]: Call `flip` directly.

File: python/mxnet/ndarray/numpy/_op.py [hunk 4]
```diff
@@ -4741,6 +4742,120 @@ def flip(m, axis=None, out=None):
...
+    >>> np.all(np.fliplr(A) == A[:,::-1,...])
+    array(True)
+    """
+    from ...numpy import ndarray
+    if isinstance(m, numeric_types):
+        return _np.flip(m, 1)
+    elif isinstance(m, ndarray):
+        return _npi.flip(m, 1)
+    else:
+        raise TypeError('type {} not supported'.format(str(type(m))))
```
Comment (reminisce) [inline python/mxnet/ndarray/numpy/_op.py]: Call `flip`.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 5]
```diff
@@ -4520,6 +4520,64 @@ def flip(m, axis=None, out=None):
...
+    out : array_like
+        A view of `m` with the rows reversed.  Since a view is
+        returned, this operation is :math:`\mathcal O(1)`.
+    """
+    if isinstance(m, numeric_types):
+        return _np.flip(m, 0)
+    elif isinstance(m, _Symbol):
+        return _npi.flip(m, 0)
+    else:
+        raise TypeError('type {} not supported'.format(str(type(m))))
```
Comment (reminisce) [inline python/mxnet/symbol/numpy/_symbol.py]: Call `flip`.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 6]
```diff
@@ -4520,6 +4520,64 @@ def flip(m, axis=None, out=None):
...
+    f : ndarray
+        A view of `m` with the columns reversed.  Since a view
+        is returned, this operation is :math:`\mathcal O(1)`.
+    """
+    if isinstance(m, numeric_types):
+        return _np.flip(m, 1)
+    elif isinstance(m, _Symbol):
+        return _npi.flip(m, 1)
+    else:
+        raise TypeError('type {} not supported'.format(str(type(m))))
```
Comment (reminisce) [inline python/mxnet/symbol/numpy/_symbol.py]: Call `flip`.

File: src/operator/numpy/np_matrix_op-inl.h [hunk 7]
```diff
@@ -498,7 +498,7 @@ void NumpyFlipForward(const nnvm::NodeAttrs& attrs,
   std::vector<index_t>  trailing_(axistemp.ndim());
   index_t flip_index = 0;
   for (int axis : axistemp) {
-    CHECK_LT(axis, ishape.ndim());
+    CHECK_LT(axis, ishape.ndim()) << "Input must be >= 2-d";
```
Comment (reminisce) [inline src/operator/numpy/np_matrix_op-inl.h]: The error message seems not accurate.

**Pull request merged**